File: android/core/designsystem/src/main/kotlin/com/riskdetectedan/core/designsystem/isg/NovaCompanyDestination.kt
package com.riskdetectedan.core.designsystem.isg

import androidx.compose.runtime.*
import kotlinx.coroutines.CancellationException
import kotlinx.coroutines.currentCoroutineContext
import kotlinx.coroutines.ensureActive
import java.util.UUID

/** The caller injects the existing repository adapter; this module knows no SDK or endpoint. */
@Composable
fun NovaCompanyDestination(host: NovaSessionHost, loadCompanies: suspend (Boolean) -> List<NovaOwnedCompany>,
                           includeArchived: Boolean = false, onSelect: (UUID) -> Unit, onBack: () -> Unit) {
    var state by remember { mutableStateOf(NovaCompanyListState()) }
    var refresh by remember { mutableStateOf(UUID.randomUUID()) }
    val currentHost by rememberUpdatedState(host)
    val currentLoader by rememberUpdatedState(loadCompanies)
    val currentArchiveScope by rememberUpdatedState(includeArchived)
    val content = state.content(host, includeArchived)
    val epoch = host.navigation.epoch
    LaunchedEffect(epoch, includeArchived, refresh) {
        state = state.begin(currentHost, includeArchived)
        val ticket = state.pending ?: return@LaunchedEffect
        try {
            val rows = currentLoader(includeArchived)
            currentCoroutineContext().ensureActive()
            state = state.complete(ticket, rows, currentHost)
        } catch (cancelled: CancellationException) {
            state = state.cancel(ticket, currentHost)
            throw cancelled
        } catch (_: Exception) {
            currentCoroutineContext().ensureActive()
            state = state.fail(ticket, currentHost)
        }
    }
    key(epoch) { // Reset rememberSaveable search/focus when the account/permission scope changes.
    NovaCompaniesScreen(companies = content.rows.map { NovaCompanyItem(it.id.toString(), it.name, it.detail) },
        isLoading = content.phase == NovaCompanyListPhase.loading || content.phase == NovaCompanyListPhase.idle,
        error = if (content.phase == NovaCompanyListPhase.failed) "Firmalar yüklenemedi. Lütfen tekrar deneyin." else null,
        isOwnedList = true,
        onSelect = { raw ->
            val id = runCatching { UUID.fromString(raw) }.getOrNull()
            val row = id?.let { state.select(it, content.requestID, currentHost, currentArchiveScope) }
            if (row != null) onSelect(row.id)
        }, onBack = { if (currentHost.isCurrent(epoch)) onBack() },
        onRetry = { if (currentHost.isCurrent(epoch)) refresh = UUID.randomUUID() })
    }
}

File: App/DesignSystem/ISG/NovaCompanyDestination.swift
import SwiftUI

/// Lifecycle-bound feature destination. The composition root injects the existing service;
/// no singleton, Auth SDK or live endpoint is pulled into the design system or QA host.
struct NovaCompanyDestination: View {
    @Binding var host: NovaSessionHost
    let loadCompanies: @MainActor (Bool) async throws -> [NovaOwnedCompany]
    var includeArchived = false
    let onSelect: (UUID) -> Void
    let onBack: () -> Void
    @State private var state = NovaCompanyListState()
    @State private var refresh = UUID()

    private struct LoadKey: Equatable { let epoch: String; let archived: Bool; let refresh: UUID }
    var body: some View {
        let current = state.content(host: host, includeArchived: includeArchived)
        let epoch = host.navigation.epoch
        NovaCompaniesScreen(companies: current.rows.map { .init(id: $0.id.uuidString.lowercased(), name: $0.name, detail: $0.detail) },
            isLoading: current.phase == .loading || current.phase == .idle,
            error: current.phase == .failed ? "Firmalar yüklenemedi. Lütfen tekrar deneyin." : nil,
            isOwnedList: true,
            onSelect: { raw in
                guard let id = UUID(uuidString: raw), let selected = state.select(id, requestID: current.requestID, host: host, includeArchived: includeArchived) else { return }
                onSelect(selected.id)
            }, onBack: { if host.isCurrent(epoch) { onBack() } },
            onRetry: { if host.isCurrent(epoch) { refresh = UUID() } })
        .id(epoch) // The previous account's search text/focus must not survive the host boundary.
        .task(id: LoadKey(epoch: epoch, archived: includeArchived, refresh: refresh)) {
            guard let ticket = state.begin(host: host, includeArchived: includeArchived) else { return }
            do {
                let rows = try await loadCompanies(includeArchived)
                try Task.checkCancellation()
                state.complete(ticket, rows: rows, host: host)
            } catch is CancellationError {
                state.cancel(ticket, host: host)
            } catch {
                if Task.isCancelled { state.cancel(ticket, host: host) }
                else { state.fail(ticket, host: host) }
            }
        }
    }
}
